include/graaflib/algorithm/shortest_path.tpp

```diff
@@ -207,7 +207,19 @@ bellman_ford_shortest_paths(const graph<V, E, T>& graph,
       }
     }
   }
-
+  // Negative cycle detection by doing an additional pass in the graph
+  for (const auto& [edge_id, edge] : graph.get_edges()) {
+    const auto [u, v]{edge_id};
+    WEIGHT_T weight = get_weight(edge);
+    if (shortest_paths[u].total_weight !=
+            std::numeric_limits<WEIGHT_T>::max() &&
+        shortest_paths[u].total_weight + weight <
+            shortest_paths[v].total_weight) {
```

Comment:
Since we are using the same predicate in the loop above, what do you think about extracting it? Maybe we could do so to a lambda in the current function body. i.e.

```c++
const auto found_shorter_path{[&shortest_paths](edge_id_t edge_id, const E& edge) {
  const auto weight{get_weight(edge);
  const auto [u, v]{edge_id};
  return shortest_paths[u].total_weight != std::numeric_limits<WEIGHT_T>::max() &&
    shortest_paths[u].total_weight + weight < shortest_paths[v].total_weight;
}};

// Relax edges for |V| - 1 iterations
for (std::size_t i = 1; i < graph.vertex_count(); ++i) {
  for (const auto& [edge_id, edge] : graph.get_edges()) {
    if (found_shorter_path(edge_id, edge)) {
      // Update the shortest path to the target of the edge...
    }
  }
}

// Negative cycle detection by doing an additional pass in the graph
for (std::size_t i = 1; i < graph.vertex_count(); ++i) {
  for (const auto& [edge_id, edge] : graph.get_edges()) {
    if (found_shorter_path(edge_id, edge)) {
      // throw...
    }
  }
}
```